vike/client/client-routing-runtime/history.ts

```diff
@@ -97,27 +100,27 @@ function replaceHistoryState(state: HistoryState, url?: string) {
   window.history.replaceState(state, '', url ?? /* Passing `undefined` chokes older Edge versions */ null)
 }
 function pushHistoryState(state: HistoryState, url: string) {
-  pushStateOriginal(state, '', url)
+  // Call history.pushState so that any other libraries that monkey patch pushState after Vike does will still run
+  window.history.pushState(state, '', url)
 }
 
 function monkeyPatchHistoryPushState() {
-  globalObject.pushStateOriginal = globalObject.pushStateOriginal ?? window.history.pushState
-  window.history.pushState = (stateFromUser: unknown = {}, ...rest) => {
-    assertUsage(
-      null === stateFromUser || undefined === stateFromUser || isObject(stateFromUser),
-      'history.pushState(state) argument state must be an object'
-    )
-    const state: HistoryState = {
-      scrollPosition: getScrollPosition(),
-      timestamp: getTimestamp(),
-      ...stateFromUser,
-      // Don't allow user to overwrite triggeredBy as it would break Vike's handling of the 'popstate' event
-      triggeredBy: 'user'
+  window.history.pushState = new Proxy(window.history.pushState, {
+    apply: (target, thisArg, [stateFromUser, unused, url]) => {
+      assertUsage(
+        null === stateFromUser || undefined === stateFromUser || isObject(stateFromUser),
+        'history.pushState(state) argument state must be an object'
+      )
+      const state: HistoryState = isVikeHistoryState(stateFromUser)
+        ? stateFromUser
```

Comment:
so technically, in userland I could call `history.pushState({timestamp: 1, scrollPosition: 2, triggeredBy: "me"})` and we would trust it, so this is not the most robust system.

We could create a `Symbol` in this scope if we really want to be sure that no userland dev can accidentally bypass this check.